### PR TITLE
AWS Lambda hosting sample  - dispatching a message outside of a message handler

### PR DESCRIPTION
**Pull request by cquirosj**
New AWS Lambda hosting sample for dispatching a message outside of a message handler. Similar to what the Azure Functions sample does.

### CODE REVIEW

**Comment by cquirosj** [comment]
@mauroservienti could you give us a review? 

In this PR, we cloned the original AWS Lambda sample and added a new lambda function that is triggered by an HTTP call. This function takes care of dispatching a TriggerMessage.

(You can see the differences between the original sample and the new one in [this commit](https://github.com/Particular/docs.particular.net/pull/6155/commits/e53033d83886c6e0ba575233d11c5035a6728153))

Notice how, although the endpoint is wired to route `TriggerMessage` to itself, it won't process the message that was just sent to the queue unless a separate call to the .Process() method is done. This requires [a separate lambda function that is triggered by SQS events](https://github.com/Particular/docs.particular.net/pull/6155/files#diff-cab69b5b03867530e2453dcd198e0488d8c6fba1d6e525520c5685b2ecb2da34R23-R29) to "forward" the SQS messages to the NSB Pipeline. I suppose this is required because the lambda function is single-threaded? Therefore the NSB endpoint won't "listen" to messages being pushed to the queue.

File: samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.SQSTrigger/aws-lambda-tools-defaults.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "All the command line options for the Lambda command can be specified in this file."
+  ],
+  "profile": "default",
+  "region": "us-west-1",
+  "configuration": "Debug",
+  "framework": "net6.0",
+  "template": "serverless.template",
+  "template-parameters": "",
+  "stack-name": "awslambda-nsb-modified-sample-with-http-annotation",
+  "s3-bucket": "cq-cloudxp-tf-154-s3-west"
```
Comment (cquirosj) [inline samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.SQSTrigger/aws-lambda-tools-defaults.json:15]:
```suggestion
  "template-parameters": ""
```

File: samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.Sender/Program.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+using NServiceBus.Logging;
+
+class Program
+{
+  static async Task Main(string[] args)
+  {
+    Console.Title = "AwsLambda.Sender";
+
+    var endpointConfiguration = new EndpointConfiguration("AwsLambda.Sender");
+    endpointConfiguration.SendFailedMessagesTo("ErrorAwsLambdaSQSTrigger");
```
Comment (mauroservienti) [inline samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.Sender/Program.cs:14]: Why not just "error"?

File: samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.SQSTrigger/Function.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        #endregion
+
+        #region EndpointSetup
+
+        private static readonly IAwsLambdaSQSEndpoint serverlessEndpoint = new AwsLambdaSQSEndpoint(context =>
+        {
+            var endpointConfiguration = new AwsLambdaSQSEndpointConfiguration("AwsLambdaSQSTrigger");
+            endpointConfiguration.UseSerialization<NewtonsoftJsonSerializer>();
+
+            var transport = endpointConfiguration.Transport;
```
Comment (mauroservienti) [inline samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.SQSTrigger/Function.cs:59]: it looks like this is not used

File: samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.SQSTrigger/Function.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            var routing = endpointConfiguration.RoutingSettings;
+
+            routing.RouteToEndpoint(typeof(TriggerMessage), "AwsLambdaSQSTrigger");
+            routing.RouteToEndpoint(typeof(BackToSenderMessage), "AwsLambda.Sender");
+
+            var advanced = endpointConfiguration.AdvancedConfiguration;
+            advanced.SendFailedMessagesTo("ErrorAwsLambdaSQSTrigger");
```
Comment (mauroservienti) [inline samples/hosting/aws-lambda-sqs/SQSLambda_2/AwsLambda.SQSTrigger/Function.cs:66]: As below, why not just "error"?

**Comment by mauroservienti** [comment]
> Notice how, although the endpoint is wired to route `TriggerMessage` to itself, it won't process the message that was just sent to the queue unless a separate call to the .Process() method is done. This requires [a separate lambda function that is triggered by SQS events](https://github.com/Particular/docs.particular.net/pull/6155/files#diff-cab69b5b03867530e2453dcd198e0488d8c6fba1d6e525520c5685b2ecb2da34R23-R29) to "forward" the SQS messages to the NSB Pipeline. I suppose this is required because the lambda function is single-threaded? Therefore the NSB endpoint won't "listen" to messages being pushed to the queue.

I'm not following. Shall we get on a quick call?

**Comment by cquirosj** [comment]
> > Notice how, although the endpoint is wired to route `TriggerMessage` to itself, it won't process the message that was just sent to the queue unless a separate call to the .Process() method is done. This requires [a separate lambda function that is triggered by SQS events](https://github.com/Particular/docs.particular.net/pull/6155/files#diff-cab69b5b03867530e2453dcd198e0488d8c6fba1d6e525520c5685b2ecb2da34R23-R29) to "forward" the SQS messages to the NSB Pipeline. I suppose this is required because the lambda function is single-threaded? Therefore the NSB endpoint won't "listen" to messages being pushed to the queue.
> 
> I'm not following. Shall we get on a quick call?

Yes, please. I will try to find a spot in your calendar

File: samples/hosting/aws-lambda-http-sqs/SQSLambda_1/AwsLambda.Messages/SQSTriggerMessage.cs [hunk 5]
```diff
@@ -0,0 +1,5 @@
+﻿using NServiceBus;
+
+public class SQSTriggerMessage : IMessage
```
Comment (cquirosj) [inline samples/hosting/aws-lambda-http-sqs/SQSLambda_1/AwsLambda.Messages/SQSTriggerMessage.cs:3]: I think I see the intent of why this was renamed, however, all the other messages are still being handled by SQS and they don't have the SQS "prefix" in their name. Although this is a sample and is not production code and this sample is very specific to SQS. Have we considered the implications of showing a practice of marrying the message type names to the underlying transport?

**Comment by cquirosj** [comment]
Closing in favor of https://github.com/Particular/docs.particular.net/pull/6163

**Pull request closed without merging**